### ISSUE TITLE
Mothership/Strategical game improvement

### ISSUE THREAD

**Issue by JZ0125** [Anuken/Mindustry-Suggestions#1442]
**### Apparently too many people can not handle a peaceful discussion on major game mechanic differences.

I for one, really admires all the work Anuke has put into this game. I just feel this V6 jump has much more potential, and it is indeed a big drift from previous versions in many aspects. From my limited experiences sealing with user interaction and game mechanics, I feel that an aggressive strategical layer can keep the game more interested, even though much more complicated. Having an aggressive strategical layer doesnt need to make the game hard, but it does keep the challenge level up and the players engaged and motivated.

So, if any of you really dislikes my idea, please know that I'm fine with you say "Hey, I dont think this will fit the game." Just please avoid being a d and start with stuff like "what a f dumb PoS this is". If you do start with the foul mouthed option, here is my reply "what a simple minded drone you are to not want to think about higher level mechanics!"

I'm just leaving this proposal here in case if anyone do want to have a good discussion about it. It is a very crude idea based on my initial feeling about V6, feel free to expand on it. Maybe one day I will make my own game with this idea.**

**Describe the content or mechanics you are proposing.**
Mothership is your mobile base of operation
Mothership can be upgraded through research and construction.
Mothership has a hanger bay for cores and shuttles
Hanger bay can be expanded.
Cores and shuttles needs resources to be built.
Cores are your forward operating base as we have now.
Shuttles deliver/fetches resources from/to mothership, even if we dont have the launchpad tech.
Shuttles have limited range/speed, but can be upgraded.
Mothership hovers over a sector, thus you can only collect resources from limited sectors at any time
Mothership can provide different support from orbit, based on researches. Such as orbital strike or maybe even terraforming.
Enemy has a mothership/motherships as well. 
Units can be launched to the mothership and dropped into another sector.
Ship to ship battle?
Strategical sector denial system?

**Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.**

V6 Planetary campaign adds a lot of sectors, but I feel there are too many for resource collection purposes. Plus I feel that the research system and strategical game has poor progression as of now. I find that there is a lack of incentive to work fast and build all the sectors very well. The launch pad was gated behind a long chain of mandatory sectors to prevent early massing of resources, and render other procedural sectors unnecessary.

By adding mothership system, the game can create more tension between the player and the enemy forces. It will expand the research tree beyond blueprints, making use of the tones of resources we gather later in the game, while create a reasonable limit for early game resource gathering. It would also create strategical importance to certain sectors for defensive and offensive purposes, instead of just being production center after capture.

In short, I'm proposing some strategical competition system like the carrier command. You fight with the enemy to capture the planet, make it possible to actually lose the campaign. You gain advantage by researching strategical capabilities, and gain blueprints by doing preset sectors, which leads to more strategical abilities. Make the game into a tower defense/offense game where you have to build your own factories and global logistics.

**Before making this issue, place an `X` in the boxes below to confirm that you have acknowledged them.** *Failure to do so may result in your request being closed automatically.*

1. - [x] I have done a quick search in the list of suggestions to make sure this has not been suggested yet.
2. - [x] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
3. - [x] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
4. - [x] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by Humanoid-X** [comment]
TOO OP

**Comment by Arackulele** [comment]
I dont think this would be ''too op'' or ''too easy'' but i think its just a stupid idea that doesnt fit mindustry

**Comment by JZ0125** [comment]
> I dont think this would be ''too op'' or ''too easy'' but i think its just a stupid idea that doesnt fit mindustry

Well, you could just strike out "stupid".

**Comment by JZ0125** [comment]
> TOO OP

I dont think it would be if it is balanced right.

**Comment by Kyllingene** [comment]
The problem with this is that it completely changes everything about Mindustry, including the new update. First, you no longer simply arrive in a sector, you have to see if you have sufficient resources, and if you don't, it's harder to get more. Also, the idea of losing a campaign sounds quite aggravating.

 - How would this mesh with the recapture mechanic?
 - How would this mesh with the global materials mechanic?
 - How would ship-to-ship battle work in a Mindustry-esc way?
 - How would terraforming work?
 - How would you balance a giant space beam?

I believe you are looking for a game more like StarCraft. Go check it out.

**Comment by JZ0125** [comment]
> The problem with this is that it completely changes everything about Mindustry, including the new update. First, you no longer simply arrive in a sector, you have to see if you have sufficient resources, and if you don't, it's harder to get more. Also, the idea of losing a campaign sounds quite aggravating.
> 
> * How would this mesh with the recapture mechanic?
> * How would this mesh with the global materials mechanic?
> * How would ship-to-ship battle work in a Mindustry-esc way?
> * How would terraforming work?
> * How would you balance a giant space beam?
> 
> I believe you are looking for a game more like StarCraft. Go check it out.

I'm pretty sure SC does not have this feature. I played it a couple dozen times already decades ago.

As I said, this setup would be more like Carrier Command: Gaea Mission (I know that is not the perfect game, but just trying to give a similar strategy system setup).  So my answer to your questions are:

- How would this mesh with the recapture mechanic?
    An actively expanding enemy force would keep the player on their toes, you would be fighting for planetary domination, instead of taking your time just expand throughout the planet. Instead of just recapturing sectors taken by you, the enemy faction would be capturing neutral sectors as well

- How would this mesh with the global materials mechanic?
    You have to build every sector well to provide the resource generation. You still have a global materials storage, just it would be rather limited at the beginning without all the researches. The research system can expand your total storage/delivery capability between mothership/sectors and sectors.

- How would ship-to-ship battle work in a Mindustry-esc way?
   Maybe build your ship block by block like your do in a sector? have some ship specific subsystems and weapons? Have spaceborne units to do a space fleet battle?

- How would terraforming work?
    Late game tech, just a time consuming way to get rid of some of the impassable terrain? So we can have a nice flat surface to build our mega factories?

- How would you balance a giant space beam?
    I dont know right now, like a GDI's ion cannon in C&C? it needs cool down and uplink for sure. And there would be defense systems like shield projectors to protect the base.

Anyway, this is just me seeing this V6 update as a very ground breaking change to the game's campaign setup. Yet I feel there could be more for V6's planetary scope setup. I mean, for pre V6, there is not dozens of sectors to play, and it was always a survival game in that one sector and trying to get a highscore. I agree that having a possibility to lose the campaign can be undesirable for some of the current players, but at this point, I was more bored by the repetitive sector capture and forget, feeling like wasting a good ton of hours and not saying "HEYYYY, I industrialized the entire surface of the planet with state of the art factories and defense systems!!!!!!!"

TL:DR, Pre V6 setup is geared towards metagame between preset maps and perfecting a design on one specific map to get the highest possible score. But for V6 setup, it is more of capturing sectors with limited waves. And either with perfect setup or not to generate the resources, you can always wait a while to have enough to load up the core with enough resources so that you can just plop down in a sector, build high-end defense systems, and have the launchpads to ship ammo to the core and transport it to them. For many of the sectors, it lacks the necessity to perfect the on site procurement like I would do in pre V6 game.

**Comment by Kyllingene** [comment]
The main problem I have with this is that it completely changes the campaign in most ways. Instead of a strategical *building* game, its just a strategy game in general. Anyway, here are my specific counterpoints:

>An actively expanding enemy force would keep the player on their toes, you would be fighting for planetary domination, >instead of taking your time just expand throughout the planet. Instead of just recapturing sectors taken by you, the enemy >faction would be capturing neutral sectors as well
The idea of Mindustry is not active enemies to overcome. If Anuken wanted that, they would have made attack maps more common in the campaign. This would be a great mod (if it's possible as a normal mod, otherwise I wish you luck), but I honestly do not believe that Mindustry would still be Mindustry with these updates. Not bad ideas in general, just a different feel. You yourself said it would be like Carrier Command, which is VERY different from Mindustry and should never be similar.

>An actively expanding enemy force would keep the player on their toes, you would be fighting for planetary domination, >instead of taking your time just expand throughout the planet. Instead of just recapturing sectors taken by you, the enemy >faction would be capturing neutral sectors as well
Again, passive expansion is the idea of Mindustry, not fighting or you lose. I for one like Mindustry as-is, and would probably stop playing (except for logic every so often) because it would just not be Mindustry. I am not looking for Carrier Command in this game.

>You have to build every sector well to provide the resource generation. You still have a global materials storage, just it would >be rather limited at the beginning without all the researches. The research system can expand your total storage/delivery >capability between mothership/sectors and sectors.
Wait, so what about the shuttles? Are they only for building cores and more shuttles? Also, I believe that a global cap on resources might not be great. Why not? Because either you would have to devise a system where global and local storage are separate but intertwined, OR limit local resources (a terrible idea!).

>Maybe build your ship block by block like your do in a sector? have some ship specific subsystems and weapons? Have >spaceborne units to do a space fleet battle?
Ok, so first of all: how do you balance enemy motherships? It's hard enough on ground. There, they just build schematics, but making it equal to the advancement of the player is nearly impossible, which you would have to do with motherships. Also, what do you mean by "ship specific subsystems"? Also, what would be the point of new weapons? What's wrong with the current ones? For another thing, unit battles are pretty boring. They shoot each other. And finally, do you win when you destroy the enemy mothership? What qualifies as "destroying"?

>Late game tech, just a time consuming way to get rid of some of the impassable terrain? So we can have a nice flat surface to >build our mega factories?
1: "time consuming". Why bother with that when you could capture another sector? Unless you have them all. In which case, you are defeating the entire purpose of your update by making this.

All in all, it's a cool concept, but not good for Mindustry as it is.

**Comment by XxsuperdeadlysniperxX** [comment]
@JZ0125 if you want a game make one yourself instead of making changes that would turn it into a whole other game bud 

This is a passive aggressive approach to this because why the hell would you want anuke to throw everything thing he made out the window just to work on a non mindustry game 

Im saying "DUMB IDEA WHY TRY?"

**Comment by XxsuperdeadlysniperxX** [comment]
Tho if it was its own game it would be cool

**Comment by JZ0125** [comment]
> @JZ0125 if you want a game make one yourself instead of making changes that would turn it into a whole other game bud 
> 
> This is a passive aggressive approach to this because why the hell would you want anuke to throw everything thing he made out the window just to work on a non mindustry game 
> 
> Im saying "DUMB IDEA WHY TRY?"

Look, I thought we can objectively discuss our thoughts here. I really appreciate all the work anuke has done and the game is good.

That idea is just sth popped into my mind as I play, and I thought it would be cool to have it, and let me put it on here just in case. But apparently you people just cant fing discuss it politely. So f off you dipshits. At least I paid for the game. Wonder how many of you have done that?

**Comment by itcannotbe** [comment]
This thread has been dormant for over a month now and someone just revived it with a comment that is not even constructive in anyway, shape, or form.

**Comment by Kyllingene** [comment]
> At least I paid for the game. Wonder how many of you have done that?

Actually, I have. I am pretty darn sure XxsuperdeadlysniperxX has as well, as he has mentioned stuff about Steam servers. 

> (the rest of your comment)

I understand your frustration about sniper's toxicity, but not only are you getting a *little* overboard, you also decided to take it out on me. For no reason. If I have been toxic enough to warrant such a response, please explain how. And yes, this is a place for suggestions. You had every right to post this. It just so happens that two people in the community dislike your suggestion strongly.

**Comment by JZ0125** [comment]
> I understand your frustration about sniper's toxicity, but not only are you getting a _little_ overboard, you also decided to take it out on me. For no reason. If I have been toxic enough to warrant such a response, please explain how. And yes, this is a place for suggestions. You had every right to post this. It just so happens that two people in the community dislike your suggestion strongly.

Honestly, I didnt want to take it out on you. I admit my proposal lacks many details, as it is really a crude idea just popped into my mind as I played the game. So I would like proper feedback on that as well. In fact, I liked that you actually responded with a slew of proper questions. And it is my sincere apologies that I cant really express my ideas very well due to my lack of organized writing.

For your response, I was a bit annoyed by the fact you suggested Star Craft, as unless there is a new version of it that I am not aware of, the StarCraft I played really does not have any strategical conquer mode... It made your response feel poorly researched, and a bit arrogant.

Anyway, thank you for your reply!

**Comment by JZ0125** [comment]
> This thread has been dormant for over a month now and someone just revived it with a comment that is not even constructive in anyway, shape, or form.

LOL, I know, right? All I want is a peaceful discussion, sigh...

**Comment by JZ0125** [comment]
> Actually, I have. I am pretty darn sure XxsuperdeadlysniperxX has as well, as he has mentioned stuff about Steam servers.
Not on this page, and I'm very sorry that I really dont have the time to go check Discord and other issue pages...
Even if he has paid for the game, it does not warrant any right to be a d.
I'm sure there are people like you that has extensively contributed to this project, but it only takes a couple dummies to ruin the good reputations.

**Comment by Kyllingene** [comment]
> Not on this page, and I'm very sorry that I really dont have the time to go check Discord and other issue pages...

I don't blame you, obviously.

> I'm sure there are people like you that has extensively contributed to this project, but it only takes a couple dummies to ruin the good reputations.

I am honored that you think I have extensively contributed, but find it funny, as all I actually did was make a slight edit to the README.md. I have only been on this repo for about 2-3 months.

> For your response, I was a bit annoyed by the fact you suggested Star Craft, as unless there is a new version of it that I am not aware of, the StarCraft I played really does not have any strategical conquer mode... It made your response feel poorly researched, and a bit arrogant.

Sorry if I seemed arrogant, I really didn't mean to come across that way. What I meant is that the idea of a mothership reminded me (for some reason, I don't understand why) of something StarCraft would have. Possibly because you technically have one (as Terran, anyway), it just doesn't do anything?

> Honestly, I didnt want to take it out on you. I admit my proposal lacks many details, as it is really a crude idea just popped into my mind as I played the game... And it is my sincere apologies that I cant really express my ideas very well due to my lack of organized writing.

No apologies necessary, all that is expected is that you think about it (which you did). I guess that this would probably be better as either a (very extensive) mod, or a new game. However, the suggestion did no harm, and would be interesting to see in some capacity, somewhere. Again, not a dumb idea in any way, just an idea that would cause controversy.
